### ISSUE TITLE
Clarify behaviour of getPointAtLength() and getPathSegmentAtLength() at discontinuities.

### ISSUE THREAD

**Issue by BigBadaboom** [w3c/svgwg#282]
https://svgwg.org/svg2-draft/single-page.html#types-__svg__SVGGeometryElement__getPointAtLength
https://svgwg.org/specs/paths/#InterfaceSVGPathElement

I think we need to enhance the algorithm for `getPointAtLength()` to clarify which segment is considered current when calculating the coordinates.  Browsers have some differences in this regard.

Consider the following test:

```
<svg>
  <path id="mypath"
        d="M50,50 L150,50
           M50,100 L150,100"/>
</svg>

var mypath = document.getElementById("mypath");
var lens = [0, 100];

for (var i=0; i<lens.length; i++) {
  var p = mypath.getPointAtLength(lens[i]);
  console.log("Point at length "+lens[i]+" = "+p.x+","+p.y);
  console.log("PathSeg at length "+lens[i]+" = "+mypath.getPathSegAtLength(lens[i]));
}

```
https://jsfiddle.net/yn6t6gr0/1/

At length 100, 

Chrome returns (150,50) and path seg 1
IE11 returns (50,100) and path seg 1
Firefox returns (50,100) and path seg 1.

I'm not sure which of the two alternative points should be considered the more "correct", but browsers ought to be consistent. Also whatever coords are returned should match with the segment returned at that point.  Chrome is consistent here, but the other two are not.

If, for example, it is decided that path segments are considered to be start-inclusive and end-exclusive, then the FF and IE coords (50,100) would make sense, but then the path seg probably ought to be 2 or 3 in that case.

Finally, in the case of a length like 0, should that point be considered in the "move" path segment, or in the path segment immediately following it?  Does it make sense for it to jump to the next segment at length=+epsilon?

**Comment by AmeliaBR** [comment]
You're right, this doesn't seem to be explicitly defined anywhere.  It should be included in the implementation notes section for [Distance along a path](https://svgwg.org/svg2-draft/paths.html#DistanceAlongAPath).

We do, however, have the following detail in the [Path Directionality](https://svgwg.org/svg2-draft/paths.html#PathDirectionality) implementation notes:

>  if the given distance along the path occurs at a path segment boundary, then the direction of the path is the direction at the start of the segment at the given distance, considering each segment to be endpoint exclusive.

So, to be consistent with that, we would want the getPointAtLength method to return the start of the new segment, not the end of the old, when the distance exactly matches the boundary.

**Comment by BigBadaboom** [comment]
I agree with that.

It would then follow logically that `getPathSegmentAtLength()` would never return a "move" segment since the move is effectively already over by the time you reach it.   So in my example above, the correct return values would then be (50,100) and path seg 3 (?)

**Comment by AmeliaBR** [comment]
That makes sense re "move" segments.

I'm not sure whether there would be any implementation issues of having a special rule for the last segment / last point on the path.

**Comment by boggydigital** [comment]
Not blocking updated 2.0 CR publication - assigning 2.1 WD milestone
